inji-web/src/components/Credentials/Crendential.tsx

```diff
@@ -8,26 +8,40 @@ import {api} from "../../utils/api";
 import {CredentialProps} from "../../types/components";
 import {CodeChallengeObject, CredentialConfigurationObject} from "../../types/data";
 import {RootState} from "../../types/redux";
+import {DataShareExpiryModal} from "../../modals/DataShareExpiryModal";
+import {storage} from "../../utils/storage";
 
 export const Credential: React.FC<CredentialProps> = (props) => {
     const selectedIssuer = useSelector((state: RootState) => state.issuers);
+    const [credentialExpiry, setCredentialExpiry] = useState<boolean>(false);
     const language = useSelector((state: RootState) => state.common.language);
     const filteredCredentialConfig: CredentialConfigurationObject = props.credentialWellknown.credential_configurations_supported[props.credentialId];
     const credentialObject = getObjectForCurrentLanguage(filteredCredentialConfig.display, language);
-    return <ItemBox index={props.index}
-                    url={credentialObject.logo.url}
-                    title={credentialObject.name}
-                    onClick={() => {
-                        const state = generateRandomString();
-                        const code_challenge: CodeChallengeObject = generateCodeChallenge(state);
-                        window.open(api.authorization(selectedIssuer.selected_issuer, props.credentialWellknown, filteredCredentialConfig, state, code_challenge), '_self', 'noopener');
-                        addNewSession({
-                            selectedIssuer: selectedIssuer.selected_issuer,
-                            certificateId: props.credentialId,
-                            codeVerifier: state,
-                            state: state,
-                        });
-                    }}
-    />
+    const vcExpiryTimes = useSelector((state: RootState) => state.common.vcExpiryTimes);
+
+    const onSuccess = () => {
+        const state = generateRandomString();
+        const code_challenge: CodeChallengeObject = generateCodeChallenge(state);
+        window.open(api.authorization(selectedIssuer.selected_issuer, props.credentialWellknown, filteredCredentialConfig, state, code_challenge), '_self', 'noopener');
+        addNewSession({
+            selectedIssuer: selectedIssuer.selected_issuer,
+            certificateId: props.credentialId,
+            codeVerifier: state,
+            vcExpiryTimes: vcExpiryTimes ?? 1,
+            state: state,
+        });
+    }
+
+    return <React.Fragment>
+        <ItemBox index={props.index}
+                        url={credentialObject.logo.url}
+                        title={credentialObject.name}
+                        onClick={() => setCredentialExpiry(true)}/>
+        {credentialExpiry && <DataShareExpiryModal
```

Comment:
Move `<DataShareExpiryModal` for better indentation to read components

Reply:
Structured properly